runtime/near-vm-logic/src/logic.rs

```diff
@@ -264,6 +264,66 @@ impl<'a> VMLogic<'a> {
             .collect())
     }
 
+    // ###################################
+    // # String reading helper functions #
+    // ###################################
+
+    /// Helper function to read and return utf8-encoding string.
+    /// If `len == u64::MAX` then treats the string as null-terminated with character `'\0'`.
+    ///
+    /// # Errors
+    ///
+    /// * If string extends outside the memory of the guest with `MemoryAccessViolation`;
+    /// * If string is not UTF-8 returns `BadUtf8`.
+    fn get_utf8_string(&mut self, len: u64, ptr: u64) -> Result<String> {
+        let mut buf;
+        if len != std::u64::MAX {
+            if len > self.config.max_log_len {
+                return Err(HostError::BadUTF8.into());
+            }
+            buf = Self::memory_get(self.memory, ptr, len)?;
+        } else {
+            buf = vec![];
+            for i in 0..=self.config.max_log_len {
+                if i == self.config.max_log_len {
+                    return Err(HostError::BadUTF8.into());
+                }
```

Comment:
`\0` counts towards max_log_len?

Reply:
Good point. Moved the check below null-termination check.